base/MAPL_XYGridFactory.F90

```diff
@@ -894,4 +896,36 @@ subroutine file_has_corners(this,rc)
       _RETURN(_SUCCESS)
    end subroutine
 
+   subroutine add_mask(this,grid,rc)
+      class(XYGridFactory), intent(in) :: this
+      type(ESMF_Grid), intent(inout) :: grid
+      integer, intent(out), optional :: rc
+
+      integer(ESMF_KIND_I4), pointer :: mask(:,:)
+      real(ESMF_KIND_R8), pointer :: fptr(:,:)
+      integer :: i,j,status
+      type(ESMF_VM) :: vm
+      integer :: has_undef, local_has_undef
+
+      call ESMF_GridGetCoord(grid, coordDim=1, localDE=0, &
+         staggerloc=ESMF_STAGGERLOC_CENTER, &
+         farrayPtr=fptr, rc=status)
+      local_has_undef = 0
+      if (any(fptr == MAPL_UNDEF)) local_has_undef = 1
+      call ESMF_VMGetCurrent(vm,_RC)
+      call ESMF_VMAllFullReduce(vm, [local_has_undef], has_undef, 1, ESMF_REDUCE_MAX, _RC) 
+      if (has_undef == 0) then
+         _RETURN(_SUCCESS)
+      end if
```

Comment:
Common pattern so I made a macro for this.   There is also `_RETURN_UNLESS(cond)`
```suggestion
_RETURN_IF(has_undef == 0)    
```